pyschism/forcing/hycom/hycom2schism.py

```diff
@@ -727,104 +763,207 @@ def fetch_data(self, outdir: Union[str, os.PathLike], hgrid, vgrid, start_date,
+        logger.info(f'filename is {foutname}')
+        ds = xr.open_dataset(url_uv)
+        ds = convert_longitude(ds)
+        ds = ds.rename_dims({'lon':'xlon'})
+        ds = ds.rename_dims({'lat':'ylat'})
+        ds = ds.rename_vars({'lat':'ylat'})
+        ds = ds.rename_vars({'lon':'xlon'})
+        ds.to_netcdf(foutname, 'w', 'NETCDF3_CLASSIC', unlimited_dims='time')
+        ds.close()
+        return foutname
+
+    def fetch_st(self, date, fmt='schism', output_path: PathLike = None):
+
+        allowed_fmts = ['schism', 'hycom']
+        if fmt not in allowed_fmts:
+            raise ValueError(f"Argument `fmt` must be one of {', '.join(allowed_fmts)}, but got: {fmt}.")
+        if fmt == 'hycom':
+            return self.fetch_hycom(
+                    date,
+                    output_path=output_path,
+                    surf_el=False,
+                    water_temp=True,
+                    salinity=True,
+                    water_u=False,
+                    water_v=False
+                    )
+
+        foutname = pathlib.Path(os.getcwd()) / f'ST_{date.strftime("%Y%m%d")}.nc' if output_path is None \
+                else pathlib.Path(output_path)
+        database = get_database(date)
+        time_idx, lon_idx1, lon_idx2, lat_idx1, lat_idx2, x2, y2 = get_idxs(date, database, self.bbox)
+        url_ts = f'https://tds.hycom.org/thredds/dodsC/{database}?lat[{lat_idx1}:1:{lat_idx2}],' + \
+            f'lon[{lon_idx1}:1:{lon_idx2}],depth[0:1:-1],time[{time_idx}],' + \
+            f'water_temp[{time_idx}][0:1:39][{lat_idx1}:1:{lat_idx2}][{lon_idx1}:1:{lon_idx2}],' + \
+            f'salinity[{time_idx}][0:1:39][{lat_idx1}:1:{lat_idx2}][{lon_idx1}:1:{lon_idx2}]'
+
+        logger.info(f'filename is {foutname}')
+
+        ds = xr.open_dataset(url_ts)
+
+        temp = ds.water_temp.values
+        salt = ds.salinity.values
+        dep = ds.depth.values
+
+        #convert in-situ temperature to potential temperature
+        ptemp = ConvertTemp(salt, temp, dep)
+
+        #drop water_temp variable and add new temperature variable
+        ds = ds.drop('water_temp')
+        ds['temperature']=(['time','depth','lat','lon'], ptemp)
+        ds.temperature.attrs = {
+            'long_name': 'Sea water potential temperature',
+            'standard_name': 'sea_water_potential_temperature',
+            'units': 'degC'
+        }
+
+        #ds.assign(water_temp2=ptemp)
+        #ds.assign.attrs = ds.water_temp.attrs
+
+        ds = convert_longitude(ds)
+        ds = ds.rename_dims({'lon':'xlon'})
+        ds = ds.rename_dims({'lat':'ylat'})
+        ds = ds.rename_vars({'lat':'ylat'})
+        ds = ds.rename_vars({'lon':'xlon'})
+        ds.to_netcdf(foutname, 'w', unlimited_dims='time', encoding={'temperature':{'dtype': 'h', '_FillValue': -30000.,'scale_factor': 0.001, 'add_offset': 20., 'missing_value': -30000.}})
+        ds.close()
+        return foutname
+
+    def fetch_data(
```

Comment:
This function looks very different from the original, but works in exactly the same way (these changes are not API breaking). However, it is much more reusable and flexible in this way.